### PR TITLE
Zingo sync calculate nullifiers and positions

### PR DESCRIPTION
**Pull request by Oscar-Pepper**
ontop of #1317

### CODE REVIEW

File: zingo-sync/src/scan.rs [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+                        .unwrap(),
+                )
+            } else {
+                let sapling_activation_height = parameters
+                    .activation_height(NetworkUpgrade::Sapling)
+                    .expect("should have some sapling activation height");
+
+                match first_block.height().cmp(&sapling_activation_height) {
+                    Ordering::Greater => {
+                        let frontiers =
```
Comment (fluidvanadium) [inline zingo-sync/src/scan.rs:94]: so we go to the trouble of downloading the frontiers, ought we to do something with them?

File: zingo-sync/src/scan.rs [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+                            client::get_frontiers(fetch_request_sender, first_block.height() - 1)
+                                .await
+                                .unwrap();
+                        (
+                            frontiers.final_sapling_tree().tree_size() as u32,
+                            frontiers.final_orchard_tree().tree_size() as u32,
+                        )
+                    }
+                    Ordering::Equal => (0, 0),
+                    Ordering::Less => panic!("pre-sapling not supported!"),
```
Comment (fluidvanadium) [inline zingo-sync/src/scan.rs:104]: return Err instead of panicking

File: zingo-sync/src/scan.rs [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+        prev_height = Some(block.height());
+        prev_hash = Some(block.hash());
+    }
+
+    Ok(())
+}
+
+// calculates nullifiers and positions for a given compact transaction
+fn calculate_nullifiers_and_positions<D, K, Nf>(
+    transaction: &CompactTx,
```
Comment (fluidvanadium) [inline zingo-sync/src/scan.rs:298]: putting the CompactTx parameter is mostly redundant

**Review by fluidvanadium — APPROVED** [review]

File: zingo-sync/src/scan.rs (hunk 1 above)
Comment (Oscar-Pepper) [reply zingo-sync/src/scan.rs:94]: yeah thats line 99 and 100

File: zingo-sync/src/scan.rs (hunk 3 above)
Comment (Oscar-Pepper) [reply zingo-sync/src/scan.rs:298]: yeah i agree this isnt optimal in this PR, i have fixed this in a following PR where we dont need to filter on txid

**Pull request merged**